edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter.Test/MqttBrokerConnectorTest.cs

```diff
@@ -248,6 +248,34 @@ public async Task SendAsyncCancelsWhenDisconnecting()
             await Assert.ThrowsAsync<TaskCanceledException>(async () => await producer.Connector.SendAsync("boo", Encoding.ASCII.GetBytes("hoo")));
         }
 
+        // Note, that this test tests a corner case scenario, because the connection between edgeHub and the broker
+        // should not be broken, and if it is, then the MQTT library should resend the message automatically, which
+        // ultimately deliver an ACK. We added an internal timeout anyway to prevent infinite waiting (and freeze a
+        // processing loop). This timeout is 10 seconds. As a result, this test completes after 10 seconds.
```

Comment:
Could we have the 10 sec timeout be configurable? Then for testing we can lower the timeout. Or if we find 10 sec to be more than enough we can lower it as well for production.

Reply:
added setting for publish-ack timeout